File: app-mvvm/src/main/java/com/xuie/imaginary/gui/gank/show/Level0Item.java
package com.xuie.imaginary.gui.gank.show;

import com.chad.library.adapter.base.entity.AbstractExpandableItem;
import com.chad.library.adapter.base.entity.MultiItemEntity;

/**
 * @author xuie
 * @date 17-8-15
 */
public class Level0Item extends AbstractExpandableItem<Level1Item> implements MultiItemEntity {
    private String type;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public int getLevel() {
        return 0;
    }

    @Override
    public int getItemType() {
        return ExpandableItemAdapter.TYPE_LEVEL_0;
    }
}

File: app-mvvm-kotlin/src/main/kotlin/com/xuie/imaginary/gui/gank/show/Level0Item.kt
package com.xuie.imaginary.gui.gank.show

import com.chad.library.adapter.base.entity.AbstractExpandableItem
import com.chad.library.adapter.base.entity.MultiItemEntity

/**
 * @author Jie Xu
 * @date 17-8-15
 */
class Level0Item : AbstractExpandableItem<Level1Item>(), MultiItemEntity {
  var type: String? = null

  override fun getLevel(): Int {
    return 0
  }

  override fun getItemType(): Int {
    return ExpandableItemAdapter.TYPE_LEVEL_0
  }
}
